app/model/Forms.scala

```diff
@@ -146,21 +149,39 @@ object Forms {
       su.setInclusiveHostCount(true)
       su.getInfo
     }
-    val whitelistedIpConstraint: Constraint[String] = Constraint({
-      whitelistedIp =>
-        Try(new SubnetUtils(whitelistedIp)) match {
-          case Failure(_) => Invalid(Seq(ValidationError("whitelistedIp.invalid")))
-          case _ =>
-            val ipAndMask = whitelistedIp.split("/")
-            if (privateNetworkRanges.exists(_.isInRange(ipAndMask(0)))) Invalid(Seq(ValidationError("whitelistedIp.invalid.private")))
-            else if (ipAndMask(1).toInt < 24) Invalid(Seq(ValidationError("whitelistedIp.invalid.range")))
-            else Valid
-        }
+    val whitelistedIpsConstraint: Constraint[String] = Constraint({
+      whitelistedIps => toSetOfWhitelistedIps(whitelistedIps).map(validateWhitelistedIp).fold(Valid)(reduceValidationResults)
     })
 
-    val form = Form(mapping(
-      whitelistedIp -> text.verifying(whitelistedIpConstraint).verifying("whitelistedIp.required", _.nonEmpty)
-    )(WhitelistedIpForm.apply)(WhitelistedIpForm.unapply))
+    def reduceValidationResults(a: ValidationResult, b: ValidationResult): ValidationResult = {
+      (a, b) match {
+        case (Valid, Valid) => Valid
+        case (Valid, i: Invalid) => i
+        case (i: Invalid, Valid) => i
+        case (Invalid(e1), Invalid(e2)) => Invalid(e1 ++ e2)
+      }
+
+    }
+
+    def validateWhitelistedIp(whitelistedIp: String): ValidationResult = {
+
+      Try(new SubnetUtils(whitelistedIp)) match {
+        case Failure(_) => Invalid(Seq(ValidationError("whitelistedIp.invalid", whitelistedIp)))
+        case _ =>
+          val ipAndMask = whitelistedIp.split("/")
+          if (privateNetworkRanges.exists(_.isInRange(ipAndMask(0)))) Invalid(Seq(ValidationError("whitelistedIp.invalid.private", whitelistedIp)))
+          else if (ipAndMask(1).toInt < 24) Invalid(Seq(ValidationError("whitelistedIp.invalid.range", whitelistedIp)))
```

Comment:
Why are we preventing the user from entering larger blocks?

Reply:
If it is legit, you could use the SubnetUtils.getInfo().getAddressCountLong() instead. May read better (but may not, to taste)

Reply:
We decided that entering a low number in the mask could expand millions of IP addresses easily, making the IP whitelisting useless. The address count method may be worth trying. The equivalent would be 256 addresses as a maximum value (2 to the power of 32 - 24)